src/util/ssa_expr.h

```diff
@@ -139,6 +139,21 @@ class ssa_exprt:public symbol_exprt
   /* Used to determine whether or not an identifier can be built
    * before trying and getting an exception */
   static bool can_build_identifier(const exprt &src);
+
+  void check(const validation_modet vm = validation_modet::INVARIANT) const
+  {
+    DATA_CHECK(!has_operands(), "SSA expression do not have operands");
+    DATA_CHECK(id() == ID_symbol, "SSA expression symbols are symbols");
+    DATA_CHECK(get_bool(ID_C_SSA_symbol), "wrong SSA expression ID");
+  }
```

Comment:
Btw, these methods have now become static methods that take the parent type in the recent changes to PR #3123. So that would then become `static void check(const exprt &expr, const validation_modet vm)`.

We should also add a case for `SSA_exprt` to `call_on_expr()` in `validate_expressions.cpp`.

Reply:
Done.